_drafts/release_versioning.md

```diff
@@ -0,0 +1,328 @@
+---
+title: Firmware Versioning
+description:
+  Firmware versioning with relelase management in mind
+author: tyler
+image: img/<post-slug>/cover.png # 1200x630
+---
+
+Release versioning might seem like a boring topic. Honestly, it should be. There should only be a couple of right ways to do versioning, and each project should pick one of the agreed-upon methods (SemVer[^semver], CalVer[^calver], etc.) that makes the most sense to the project. We don't live in this ideal world unfortunately and many projects choose to deviate from versioning standards. I'm not here to say they are wrong, but I'd like to discuss what they might be missing out on.
+
+Along with a typical release version, it is useful to include extra metadata, such as the Git revision, [GNU Build ID]({% post_url 2019-05-29-gnu-build-id-for-firmware %}), build timestamp, and other related items. With all of this in place, both machines and developers can quickly hone in on *exactly* which release is running on a given device, what the source is, where it came from, and how to rebuild it if necessary.
```

Comment:
> build timestamp

heh, torn on this one, from a reproducibility perspective, having timestamps in a build can be annoying because it will change a md5

Reply:
I'm sure you can technically generate a new timestamp if and only if something *else* changed?